Microsoft.Identity.Web/Resource/ScopesRequiredHttpContextExtensions.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License
+
+using Microsoft.AspNetCore.Http;
+using System;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Security.Claims;
+
+namespace Microsoft.Identity.Web.Resource
+{
+    public static class ScopesRequiredHttpContextExtensions
+    {
+        /// <summary>
+        /// When applied to an <see cref="HttpContext"/>, verifies that the user authenticated in the
+        /// web API has any of the accepted scopes. 
+        /// If the authentication user does not have any of these <paramref name="acceptedScopes"/>, the
+        /// method throws an HTTP Unauthorized with the message telling which scopes are expected in the token
+        /// </summary>
+        /// <param name="acceptedScopes">Scopes accepted by this web API</param>
+        /// <exception cref="HttpRequestException"/> with a <see cref="HttpResponse.StatusCode"/> set to 
+        /// <see cref="HttpStatusCode.Unauthorized"/>
+        public static void VerifyUserHasAnyAcceptedScope(this HttpContext context, params string[] acceptedScopes)
```

Comment:
>VerifyUserHasAnyAcceptedScope [](start = 27, length = 29)

This pattern requires a call to this method to be placed in each and every one of every controller methods, which is very excessive. A check for needed scopes can be done in middleware, preferably after the token has been validated.

Reply:
I disagree on this one, @kalyankrishna1 : you are assuming that the web app / web api will have only one controller / action.  I believe that real life applications have several actions, which would be each protected by their scope (at least the developer should chose)